File: mobile/lib/models/trade_session.dart
class TradeSession {
  final String id;
  final String participantAId;
  final String participantBId;
  final String? pieceAId;
  final String? pieceBId;
  final int deltaPinceaux;
  final String deltaDirection;
  final String status;
  final DateTime createdAt;
  final DateTime? completedAt;

  TradeSession({
    required this.id,
    required this.participantAId,
    required this.participantBId,
    this.pieceAId,
    this.pieceBId,
    this.deltaPinceaux = 0,
    this.deltaDirection = 'none',
    this.status = 'pending',
    DateTime? createdAt,
    this.completedAt,
  }) : createdAt = createdAt ?? DateTime.now();

  factory TradeSession.fromJson(Map<String, dynamic> json) {
    return TradeSession(
      id: json['id'] as String,
      participantAId: json['participant_a_id'] as String,
      participantBId: json['participant_b_id'] as String,
      pieceAId: json['piece_a_id'] as String?,
      pieceBId: json['piece_b_id'] as String?,
      deltaPinceaux: (json['delta_pinceaux'] as int?) ?? 0,
      deltaDirection: (json['delta_direction'] as String?) ?? 'none',
      status: (json['status'] as String?) ?? 'pending',
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'] as String)
          : DateTime.now(),
      completedAt: json['completed_at'] != null
          ? DateTime.parse(json['completed_at'] as String)
          : null,
    );
  }
}

File: BACKUP002/backend/app/models/trade_session.py
import uuid
from datetime import datetime
from sqlalchemy import Column, String, Integer, DateTime, Enum, ForeignKey
from sqlalchemy.orm import relationship
from ..database import Base
import enum


class DeltaDirection(str, enum.Enum):
    a_to_b = "a_to_b"
    b_to_a = "b_to_a"
    none = "none"


class TradeStatus(str, enum.Enum):
    pending = "pending"
    scanned_a = "scanned_a"
    scanned_b = "scanned_b"
    confirmed_a = "confirmed_a"
    confirmed_b = "confirmed_b"
    completed = "completed"
    cancelled = "cancelled"


class TradeSession(Base):
    __tablename__ = "trade_sessions"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    appointment_id = Column(String(36), ForeignKey("appointments.id"), nullable=True)
    participant_a_id = Column(String(36), ForeignKey("users.id"), nullable=False)
    participant_b_id = Column(String(36), ForeignKey("users.id"), nullable=False)
    piece_a_id = Column(String(36), ForeignKey("pieces.id"), nullable=True)
    piece_b_id = Column(String(36), ForeignKey("pieces.id"), nullable=True)
    delta_pinceaux = Column(Integer, default=0)
    delta_direction = Column(Enum(DeltaDirection), default=DeltaDirection.none)
    status = Column(Enum(TradeStatus), default=TradeStatus.pending)
    created_at = Column(DateTime, default=datetime.utcnow)
    completed_at = Column(DateTime, nullable=True)

    scan_events = relationship("ScanEvent", back_populates="trade_session")
